File: 2025-05-19/dz2.go
package main

import (
    "fmt"
    "math"
    "os"
    "strings"
)

func FileRead(name string) string {
    readFile, _ := os.ReadFile(name)
    return string(readFile)
}

// Функция ввоад числа с проверкой
func Num(txt string) string {
    var p string
    fmt.Printf("%s: ", txt)
    fmt.Scanln(&p)
    return p
}

func main() {
    // Задание 1
    mass := strings.Split(FileRead("text.txt"), "\n")
    mass2 := strings.Split(FileRead("text2.txt"), "\n")
    fmt.Println("ЗАДАНИЕ 1:\n")
    for i := 0; i < int(math.Max(float64(len(mass)), float64(len(mass2)))); i++ {
        if i >= len(mass) { fmt.Printf("text2.txt\tСтрока: %d:\n%s\n\n", i + 1, mass2[i]); continue }
        if i >= len(mass2) { fmt.Printf("text.txt\tСтрока: %d:\n%s\n\n", i + 1, mass[i]); continue }
        if mass[i] != mass2[i] { fmt.Printf("text.txt:\tСтрока: %d\n%s\n\ntext2.txt:\tСтрока: %d\n%s\n\n", i + 1, mass[i], i + 1, mass2[i]) }
    }

    // Задаине 2
    data := FileRead("text.txt")
    znach := []int{0, 0, 0}
    glass := "aeiouyауоиэыяюеё"
    soglass := "bcdfghjklmnpqrstvxxzбвгджзйклмнпрстфхцчшщ"
    sifra :="0123456789"
    for i := 0; i < len(data); i ++ {
        if strings.Contains(glass, strings.ToLower(string(data[i]))) { znach[0] += 1 }
        if strings.Contains(soglass, strings.ToLower(string(data[i]))) { znach[1] += 1 }
        if strings.Contains(sifra, string(data[i])) { znach[2] += 1 }
    }
    x := len(strings.Split(data, "\n"))
    f, _ := os.Create("dz2_result2.txt")
    defer f.Close()
    f.WriteString(fmt.Sprintf("Количество символов: %d\nКоличество строк: %d\nКоличество гласных: %d\nКоличество согласных: %d\nКоличество цифр: %d", len(data), x, znach[0], znach[1], znach[2]))

    // Задание 3
    mass = strings.Split(FileRead("text.txt"), "\n")
    mass = mass[:len(mass) - 1]
    f, _ = os.Create("dz2_result3.txt")
    defer f.Close()
    for _, line := range mass {
        f.WriteString(line + "\n")
    }

    // Задание 4
    mass = strings.Split(FileRead("text.txt"), "\n")
    i := len(mass[0])
    for y := 0; y < len(mass); y++ { if len(mass[y]) > i { i = len(mass[y]) } }
    fmt.Printf("ЗАДАНИЕ 4:\nДлинна самой длинной строки: %d\n\n", i)

    // Задание 5
    data = FileRead("text.txt")
    fmt.Println("ЗАДАНИЕ 5:")
    fmt.Printf("Встречается: %d раз\n\n", strings.Count(data, Num("Введите заданное слово")))

    // Задаие 6
    fmt.Println("ЗАДАНИЕ 6:")
    f, _ = os.Create("text.txt")
    defer f.Close()
    f.WriteString(strings.ReplaceAll(data, Num("Введите слово для замены"), Num("Введите на что заменить")))
}

File: 2025-05-19/dz2.py
#!/usr/bin/python3

def FileRead(name):
    ms = []
    with open(name, 'r') as f:
        for _ in f: ms.append(_.strip())
    return ms

# Задание 1
mass = FileRead('text.txt')
mass2= FileRead('text2.txt')
print('ЗАДАНИЕ 1:\n')
for _ in range(max(len(mass), len(mass2))):
    if _ >= len(mass): print(f'text2.txt\tСтрока: {_ + 1}:\n{mass2[_]}\n\n'); continue
    if _ >= len(mass2): print(f'text.txt\tСтрока: {_ + 1}:\n{mass[_]}\n\n'); continue
    if mass[_] != mass2[_]: print(f'text.txt:\tСтрока: {_ + 1}\n{mass[_]}\n\ntext2.txt:\tСтрока: {_ + 1}\n{mass2[_]}\n\n')

# Задаине 2
with open('text.txt', 'r') as f: data = f.read()
mass = [0] * 3
glass = 'aeiouyауоиэыяюеё'
soglass = 'bcdfghjklmnpqrstvxxzбвгджзйклмнпрстфхцчшщ'
for _ in data:
    if _.lower() in glass: mass[0] += 1
    if _.lower() in soglass: mass[1] += 1
    if _.isdigit(): mass[2] += 1
with open('dz2_result2.txt', 'w') as f:
    x = len(data.split('\n'))
    f.write(f'Количество символов: {len(data)}\nКоличество строк: {x}\nКоличество гласных: {mass[0]}\nКоличество согласных: {mass[1]}\nКоличество цифр: {mass[2]}')

# Задание 3
mass = FileRead('text.txt')
with open('dz2_result3.txt', 'w') as f:
    for _ in range(len(mass) - 1): f.write(f'{mass[_]}\n')

# Задание 4
i = len(mass[0])
for _ in mass:
    if len(_) > i: i = len(_)
print(f'ЗАДАНИЕ 4:\nДлинна самой длинной строки: {i}\n\n')

# Задание 5
with open('text.txt', 'r') as f: data = f.read()
print('ЗАДАНИЕ 5:')
print(f'Встречается: {data.count(input("Введите заданное слово: "))} раз\n')

# Задаие 6
print('ЗАДАНИЕ 6:')
with open('text.txt', 'w') as f: f.write(data.replace(input('Введите слово для замены: '), input('Введите на что заменить: ')))
